src/django/api/models.py

```diff
@@ -668,6 +670,12 @@ class Meta:
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
+    @staticmethod
+    def post_save(sender, **kwargs):
+        instance = kwargs.get('instance')
+        if instance.facility is not None:
```

Comment:
If no facility has been assigned to the FacilityListItem, then there's no FacilityIndex row to add any custom text to, so we don't need to try to index it